addons/gamification/models/gamification_karma_tracking.py

```diff
@@ -0,0 +1,100 @@
+            be able to retrieve last week karma changes accurately.
+        """
+        to_date = datetime.now() - relativedelta(months=1)
+        from_date = to_date - relativedelta(months=1)
+        self._process_consolidate(from_date, to_date, 'daily', 'monthly', '%Y-%m-01')
+
+    @api.model
+    def _consolidate_last_year(self):
+        """ Like the _consolidate_last_month we consolidate the monthly values in yearly values for the year before
+            the last one.
+        """
+        to_date = datetime.now() - relativedelta(years=1)
+        from_date = to_date - relativedelta(years=1)
+        self._process_consolidate(from_date, to_date, 'monthly', 'yearly', '%Y-01-01')
+
+    def _process_consolidate(self, from_date, to_date, old_consolidation_type, new_consolidation_type, tracking_date_format):
+        """
+        Consolidate all old_consolidation_type records which tracking_date is between from_date to to_date
+        into one new_consolidation_type record with a new tracking_date
+
+        :param from_date: the date from which the method will consolidate the records (included)
+        :param to_date: the date until which the method will consolidate the records (excluded)
+        :param old_consolidation_type: the consolidation_type of the records to consolidate
+        :param new_consolidation_type: the consolidation_type of the resulting record
+        :param tracking_date_format: the date format that will be used for the tracking_date of the resulting record
+        :return:
+        """
+        select_query = """
+                        SELECT user_id, MIN(old_value) as old_value, MAX(new_value) as new_value
+                        FROM gamification_karma_tracking 
+                        WHERE tracking_date::timestamp >= timestamp '{from_date}' 
+                        AND tracking_date::timestamp < timestamp '{to_date}'
+                        AND consolidation_type = '{consolidation_type}'
+                        GROUP BY user_id
+                        """.format(from_date=from_date, to_date=to_date, consolidation_type=old_consolidation_type)
+        self.env.cr.execute(select_query)
+        results = self.env.cr.dictfetchall()
+        if len(results) > 0:
+            for result in results:
+                result['consolidation_type'] = new_consolidation_type
+                result['tracking_date'] = from_date.strftime(tracking_date_format)
+                self.create(result)
+            delete_query = """
+                            DELETE FROM gamification_karma_tracking
+                            WHERE tracking_date::timestamp >= timestamp '{from_date}' 
+                            AND tracking_date::timestamp < timestamp '{to_date}'
+                            AND consolidation_type = '{consolidation_type}'
+                            """.format(from_date=from_date, to_date=to_date, consolidation_type=old_consolidation_type)
+            self.env.cr.execute(delete_query)
```

Comment:
Again, arguments in execute call.
You should never (ever) format SQL yourself to avoid injections.